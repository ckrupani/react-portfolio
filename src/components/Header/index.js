import React from "react";
import Navigation from "../Navigation";

function Header({ menuItems, activeItem, setActiveItem }) {
    return (
        <header data-testid="header" className="flex-row space-between px-2">
            <Navigation
                menuItems={menuItems}
                activeItem={activeItem}
                setActiveItem={setActiveItem}
            />
        </header>
    );
}

export default Header;
