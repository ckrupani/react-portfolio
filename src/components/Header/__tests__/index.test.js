import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Header from "..";

afterEach(cleanup);

describe("Header component", () => {
    const menuItems = [
        {
            name: "about",
            title: "About me",
        },
        { name: "portfolio", title: "Portfolio" },
        { name: "contact", title: "Contact" },
        {
            name: "resume",
            title: "Resume",
        },
    ];
    const activeItem = {
        name: "about",
        title: "About me",
    };
    const setActiveItem = jest.fn();

    it("renders", () => {
        render(
            <Header
                menuItems={menuItems}
                activeItem={activeItem}
                setActiveItem={setActiveItem}
            />
        );
    });

    it("matches snapshot DOM node structure", () => {
        const { asFragment } = render(
            <Header
                menuItems={menuItems}
                activeItem={activeItem}
                setActiveItem={setActiveItem}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
