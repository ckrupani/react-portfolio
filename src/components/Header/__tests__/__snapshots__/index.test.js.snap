// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <header
    class="flex-row space-between px-2"
    data-testid="header"
  >
    <h2
      class="header"
    >
      <a
        data-testid="link"
        href="/"
      >
        Chirag
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2 navActive"
        >
          <span>
            About me
          </span>
        </li>
        <li
          class="mx-2 false"
        >
          <span>
            Portfolio
          </span>
        </li>
        <li
          class="mx-2 false"
        >
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-2 false"
        >
          <span>
            Resume
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
